internal/fleet/package_resource.go

```diff
@@ -0,0 +1,120 @@
+package fleet
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/fleet"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+)
+
+func getPackageID(name, version string) string {
+	hash, _ := utils.StringToHash(name + version)
+
+	return *hash
+}
+
+func ResourcePackage() *schema.Resource {
```

Comment:
It feels like this resource may be difficult to use without a data source to list the available packages. I'm thinking about how I'd use this right now, where I'd be flicking between the Kibana UI and TF code, but what happens when the package in the repo is upgraded to a new version? 

Do you have plans to add that data source in a follow up? If not could we?

Reply:
These are great points. I've been trying to think of usability concerns (and there are a few when dealing with packages and package policies).

We didn't have explicit plans for a package data source, but we can definitely look into doing that. At a quick glance, the "list packages" endpoint for EPM seems to list the latest package versions only (it doesn't give you a full manifest of what's available in the package registry), as far as I can tell. This should be fine, though, since we really only care about the latest package version.

Reply:
Wdyt about timing here? I'm inclined to merge this and have the data sources in a follow up?

Reply:
I would agree as well. I can write a follow up issue for the package data source.